### ISSUE TITLE
🐛 Bug: auto focus and auto scroll of `ReactDataGrid` as soon as drop down opens on `SelectEditor`

### ISSUE THREAD

**Issue by madooei** [inovua/reactdatagrid#158]
- what edition are you using - community / enterprise

community

- version for `@inovua/reactdatagrid-community`or `@inovua/reactdatagrid-enterprise` -

`4.1.18`

Relevant code or config

- An editable `ReactDataGrid` with 40 columns and 10 rows has been created. 

```javascript
<ReactDataGrid
  idProperty="id"
  columns={columns}
  dataSource={dataSource}
  onEditComplete={onEditComplete}
  editable={true}
/>
```

- The (inline) editor for every column is a `SelectEditor` with the following `editorProps`

```javascript
editor: SelectEditor,
editorProps: {
  idProperty: "id",
  dataSource: ["Item 1", "Item 2", "Item 3"].map((element) => ({
    id: element,
    label: element
  })),
  collapseOnSelect: true,
  clearIcon: null
  // focusOnClick: false,
}
```

What you did:

* Opened the drop-down to select an item on any columns that require scrolling to the right. 

What happened:

* As soon as the drop-down opens, the grid scrolls back to the left all the way to column-1. 

  ![2021-06-23_02-14-13 (1)](https://user-images.githubusercontent.com/1509757/123046906-a5982c00-d3ca-11eb-8bc9-91a08be61afe.gif)

Reproduction repository:

* https://codesandbox.io/s/reactdataguide-issue158-lwpjo

Problem description:

* Effectively,  one would not be able to use the `SelectEditor` to update a cell value.

Suggested solution:

* Not sure; it seems when the `SelectEditor` is in "focus," the "parent" element (`ReactDataGrid`) also becomes in focus which causes the view to scroll to the cell at the top-left corner. 
* As a workaround, you can add the `focusOnClick: false` prop to `editorProps`. This solves the issue, but then it is not possible to trigger `onEditComplete` which means changing a cell value will not be persistent.

**Comment by madooei** [comment]
Until the maintainers solve this issue, here is a quick fix for anyone having this problem; create your own minimal select editor

```js
import React, { useState } from "react";

// Change this to look the way you want
const selectStyle = {
  width: "100%",
  height: "100%",
  margin: "0",
};

function MySelectEditor(props) {
  const { value, onChange, onComplete, dataSource, onCancel, onTabNavigation } = props;
  const [data, setData] = useState(value || "");

  const handleChange = (event) => {
    onChange && onChange(event);
    setData(event.target.value);
  };

  const handleOnBlur = (event) => {
    onComplete && onComplete(event);
  };

  const handleOnKeyDown = (event) => {
    const { key } = event;
    if (key === "Escape") {
      onCancel && onCancel(event);
    }
    if (key === "Enter") {
      onComplete && onComplete(event);
    }
    if (key === "Tab") {
      event.preventDefault();
      onTabNavigation && onTabNavigation(true, event.shiftKey ? -1 : 1);
    }
  };

  return (
    <select
      style={selectStyle}
      value={data}
      onChange={handleChange}
      onBlur={handleOnBlur}
      onKeyDown={handleOnKeyDown}
    >
      {dataSource.map((option, index) => {
        return (
          <option key={index} value={option.id}>
            {option.label}
          </option>
        );
      })}
    </select>
  );
}

export default MySelectEditor;
```

You must update the column props

```diff
- editor: SelectEditor,
+ renderEditor: (editorProps) => {
+  return <MySelectEditor {...editorProps} />;
+ },
  editorProps: {
    idProperty: "id",
    dataSource: ["Item 1", "Item 2", "Item 3"].map((element) => ({
      id: element,
      label: element
    })),
-  collapseOnSelect: true,
-  clearIcon: null
-  // focusOnClick: false,
  }
```

**Comment by inovua-admin** [comment]
Fixed in `v4.1.20`
